cosmos/operators/_async/base.py

```diff
@@ -0,0 +1,45 @@
+from abc import ABCMeta
+from typing import Any, Sequence
+
+from airflow.utils.context import Context
+
+from cosmos.config import ProfileConfig
+from cosmos.exceptions import CosmosValueError
+from cosmos.operators._async.bigquery import DbtRunAirflowAsyncBigqueryOperator
+from cosmos.operators.local import DbtRunLocalOperator
+
+# Register async operator here
+ASYNC_CLASS_MAP = {"bigquery": DbtRunAirflowAsyncBigqueryOperator}
+
+
+class DbtRunAirflowAsyncFactoryOperator(DbtRunLocalOperator, metaclass=ABCMeta):  # type: ignore[misc]
+
+    template_fields: Sequence[str] = ("project_dir",)
+
+    def __init__(self, project_dir: str, profile_config: ProfileConfig, **kwargs: Any):
+        self.project_dir = project_dir
+        self.profile_config = profile_config
+        self.async_args = None
+        if "async_args" in kwargs:
+            self.async_args = kwargs.pop("async_args")
```

Comment:
Why did we introduce this

 I remember we had lots of discussions, a few challenges in the past and overcame this in #1230. The idea of having these arguments at the same level as the other operator arguments is to be consistent with how other Cosmos Operators work.  It would be great if we could remain consistent, without exposing to users an internal implementation detail.

Reply:
Addressed in: a04da863529655cfe1e2722de92ee5d11785ec29 PTAL